plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -225,14 +241,15 @@ protected function render( $attributes, $content, $block ) {
 
 			ob_start();
 
-			remove_action( 'woocommerce_' . $product_type . '_add_to_cart', 'woocommerce_' . $product_type . '_add_to_cart', 30 );
-			/**
-			 * Trigger the single product add to cart action that prints the markup.
-			 *
-			 * @since 9.9.0
-			 */
-			do_action( 'woocommerce_' . $product->get_type() . '_add_to_cart' );
-			add_action( 'woocommerce_' . $product_type . '_add_to_cart', 'woocommerce_' . $product_type . '_add_to_cart', 30 );
+			if ( in_array( $product_type, array( ProductType::SIMPLE, ProductType::EXTERNAL, ProductType::VARIABLE, ProductType::GROUPED ), true ) ) {
+				$add_to_cart_fn = 'woocommerce_' . $product_type . '_add_to_cart';
+				/**
+				 * Trigger the single product add to cart action that prints the markup.
+				 *
+				 * @since 9.9.0
+				 */
+				do_action( 'woocommerce_' . $product->get_type() . '_add_to_cart' );
+			}
```

Comment:
We need to keep the `remove_action()` and `add_action()` so we don't run the WC core actions when rendering this block. Otherwise, the form appears twice:

![imatge](https://github.com/user-attachments/assets/544d7f13-5cfd-4993-8ea6-02baa8cdbf10)